Tests/Functional/Controller/ConnectTest.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Functional\Controller;
+
+use ShopwarePlugins\Connect\Tests\WebTestCaseTrait;
+use ShopwarePlugins\Connect\Tests\TestClient;
+use Symfony\Component\HttpFoundation\Response;
+
+class ConnectTest extends \PHPUnit_Framework_TestCase
+{
+    use WebTestCaseTrait;
+
+    /**
+     * @param Response $response
+     * @return array
+     */
+    private function handleJsonResponse(Response $response)
+    {
+        $this->assertEquals(200, $response->getStatusCode());
+        $responseData = \json_decode($response->getContent(), true);
+        $this->assertNotNull($responseData, 'Response is not valid JSON');
+
+        return $responseData;
+    }
+
+    public function test_get_article_source_ids()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+        $expectedIds = ['2', '2-123', '2-124'];
+
+        $client->request('POST', 'backend/Connect/getArticleSourceIds', ['ids' => [2]]);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertNotEmpty($responseData);
+        $this->assertEquals(1, $responseData['success']);
+        $this->assertEquals($expectedIds, $responseData['sourceIds']);
+    }
+
+    public function test_get_article_source_ids_with_export_all()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+        $expectedIds = ['2', '117', '2-123', '2-124', '117-221'];
+
+        $client->request('POST', 'backend/Connect/getArticleSourceIds', ['exportAll' => 'true']);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertNotEmpty($responseData);
+        $this->assertTrue($responseData['success']);
+        $this->assertArraySubset($expectedIds, $responseData['sourceIds']);
+    }
+
+    public function test_exporting_a_product_with_variants()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+
+        $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => ['2','2-123','2-124']]);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertTrue($responseData['success']);
```

Comment:
here it would make sense to check wether the export_status in s_plugin_connect_items of the products with the given source ids is 'update'